R/zchunk_LA114.wind-gcamusa.R

```diff
@@ -1,70 +1,83 @@
+    us_state_wind <- get_data(all_data, "gcam-usa/us_state_wind")
     A23.globaltech_capital <- get_data(all_data, "energy/A23.globaltech_capital")
     A23.globaltech_OMfixed <- get_data(all_data, "energy/A23.globaltech_OMfixed")
     A23.globaltech_OMvar <- get_data(all_data, "energy/A23.globaltech_OMvar")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # ===================================================
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # This function filters A23 tables for wind, then gathers and interpolates...
+    # ... to get a single value for wind base cost year. Note that the interpolation is...
+    # ... redundant whilst gcamusa.WIND_BASE_COST_YEAR = 2005, ...
+    # ... since 2005 is an existing data point in all A23 tables.
+    filter_gather_interp_get_cost <- function(x){
+      x %>% filter(technology == "wind") %>%
+        gather(year,value, `1971`, `1990`, `2005`, `2010`, `2015`) %>%
+        select(year,value) %>% mutate(year = as.integer(year)) %>%
+        complete(year = c(year, gcamusa.WIND_BASE_COST_YEAR)) %>%
+        mutate(value = approx_fun(year, value, rule = 2)) -> x_
+      return(filter(x_, year == gcamusa.WIND_BASE_COST_YEAR)$value)
+    }
+
+    A23.globaltech_capital %>% filter_gather_interp_get_cost -> L114.CapCost
+    A23.globaltech_OMfixed %>% filter_gather_interp_get_cost -> L114.OMFixedCost
+    A23.globaltech_OMvar %>% filter_gather_interp_get_cost -> L114.OMVarCost
+    # ^^ all above rates are in $1975
+
+    # Get fixed charge rate of capital for wind
+    filter(A23.globaltech_capital, technology=="wind")$fixed.charge.rate -> L114.FixedChargeRate
+
+    # Convert state level wind base cost data to 1975$/GJ, ...
+    # ... then compute capacity factor for the base wind turbine in each state
+    us_state_wind %>%
+      mutate(sector = "electricity generation",
+             fuel = "wind",
+             base_cost = base_cost * gcamdata::gdp_deflator(1975, 2007) / CONV_KWH_GJ,
```

Comment:
Note the `gcamdata::` prefix is unnecessary. But it does clarify the source of this function, so can leave it if you want.